vyper/semantics/types/function.py

```diff
@@ -631,6 +636,79 @@ def fetch_call_return(self, node: vy_ast.Call) -> Optional[VyperType]:
         return self.return_type
 
 
+class BuiltinFunctionT(VyperType):
+    _has_varargs = False
+    _inputs: list[tuple[str, Any]] = []
+    _kwargs: Dict[str, KwargSettings] = {}
+    _return_type: Optional[VyperType] = None
+
+    # helper function to deal with TYPE_DEFINITIONs
+    def _validate_single(self, arg: vy_ast.VyperNode, expected_type: VyperType) -> None:
+        # TODO using "TYPE_DEFINITION" is a kludge in derived classes,
+        # refactor me.
+        if expected_type == "TYPE_DEFINITION":
+            # try to parse the type - call type_from_annotation
+            # for its side effects (will throw if is not a type)
+            type_from_annotation(arg)
+        else:
+            validate_expected_type(arg, expected_type)
+
+    def _validate_arg_types(self, node: vy_ast.Call) -> None:
+        num_args = len(self._inputs)  # the number of args the signature indicates
+
+        expect_num_args: Union[int, tuple] = num_args
+        if self._has_varargs:
+            # note special meaning for -1 in validate_call_args API
+            expect_num_args = (num_args, -1)
+
+        validate_call_args(node, expect_num_args, list(self._kwargs.keys()))
```

Comment:
Last argument should be a list of valid keyword arguments. Previously, it was a dictionary of `KwargSettings`.